test/autorest/complexgroup/complexgroup_test.go

```diff
@@ -11,8 +11,16 @@ import (
 	"testing"
 )
 
-func getComplexClient(t *testing.T) *complexgroup.ComplexClient {
-	client, err := complexgroup.NewComplexClient(complexgroup.DefaultEndpoint, nil)
+func getBasicClient(t *testing.T) *complexgroup.BasicClient {
```

Comment:
The change here is simply renaming all of the ComplexClient references to BasicClient since ComplexClient was too broad.

Reply:
This client name comes from the swagger, specifically from the `operationId`.  E.g. for `basic_getValid` the client name would be `BasicClient` with a method `GetValid`.

I didn't look at the operations that have been implemented, but you might want to double-check that the client names match the swagger definition.

Reply:
Yes no worries. I already noticed that and that's the reason for the changes in this PR. So everything that was previously ComplexClient is now BasicClient and I also have a PrimitiveClient for operations like: primitive_getLong. Both are part of the complexgroup package